deployments/epinio.go

```diff
@@ -132,7 +132,27 @@ func (k Epinio) apply(c *kubernetes.Cluster, ui *termui.UI, options kubernetes.I
 		return errors.Wrap(err, "Couldn't get system_domain option")
 	}
 
-	message := "Creating Epinio server ingress"
+	// Wait for the cert manager to be present and active. It is required
+	for _, podname := range []string{
+		"webhook",
+		"cert-manager",
+		"cainjector",
+	} {
+		if err := c.WaitUntilPodBySelectorExist(ui, CertManagerDeploymentID, "app.kubernetes.io/name="+podname, k.Timeout); err != nil {
```

Comment:
Maybe `WaitForDeploymentCompleted` is better here? We should start talking about Deployments instead of Pods when it makes more sense.

Reply:
Will try locally before committing. It certainly would be simpler. I.e. no replication of internal code from the other deployment.
I may have to re-read all the helpers to see what we have now. I had not realized that we had the ability to wait for a deployment already.

Reply:
Hm. ... We might have a confusion here, because of names.
The `WaitForDeploymentCompleted` is for waiting on a kube `Deployment` resource.
This is not a `deployment` as we use it in `epinio install`.

That said, I will check if the helm chart deployed by the `cert-manager` deployment (sic) installs a kube `Deployment`.
And if not, maybe some other higher controller (`StateSet`, `ReplicaSet`, etc.) we could wait for.

Reply:
Well, 3 kube deployments

```
% k get deployment : |grep cert
kubectl get deployment --all-namespaces
cert-manager      cert-manager-cainjector                      1/1     1            1           18m
cert-manager      cert-manager                                 1/1     1            1           18m
cert-manager      cert-manager-webhook                         1/1     1            1           18m
```

Now I am wondering if we should/could do waiting for an epinio deployment purely internally.
Because the goroutine for deployment knows when it is done.

Maybe can use the WaitGroup we already have waiting for all of them to complete ?

Reply:
I mainly suggested to wait for deployment because it's is a higher level structure in Kubernetes and its status better reflects the service's availability. In most cases we are not gaining a lot though. Using the wait groups will complicate that code even more I'm afraid. Feel free to leave this as it is.

Reply:
Resolving, defered to future cleanup.